Avoid attaching security groups to NLB in unsupported regions and fix integration test

hanwen-pcluste opened the pull request:
See commits descriptions for details


### Tests
* Tests in the unsupported regions passed

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

hanwen-cluster committed:
[integ-tests] Fix AD integration test in iso regions

iso regions don't support tagging certificates. Therefore, the condition in IAM policy is too restrictive in iso reginos

Signed-off-by: Hanwen <hanwenli@amazon.com>

hanwen-cluster committed:
[integ-tests] Fix AD integration tests for regions with FSx OpenZFS and Ontap

Signed-off-by: Hanwen <hanwenli@amazon.com>

hanwen-cluster committed:
Avoid attach security groups to NLB in unsupported regions

NLB with security groups is a new feature added in 2023. However, the feature is not supported in ADC

Signed-off-by: Hanwen <hanwenli@amazon.com>

gmarciani commented on cli/src/pcluster/constants.py:
```diff
@@ -237,6 +237,7 @@ class Feature(Enum):
     SLURM_DATABASE = "SLURM Database"
     CLUSTER_HEALTH_METRICS = "Cluster Health Metrics"
     CAPACITY_BLOCK = "Capacity Block"
+    NLB_SECURITY_GRROUP = "NLB Security Group"
```
1. typo: GRROUP => GROUP
1. The string is customer facing, so I suggest to expand the acronym NLB to "Network Load Balancer"

hanwen-pcluste replied:
Done

gmarciani commented on cli/src/pcluster/constants.py:
```diff
@@ -249,6 +250,7 @@ class Feature(Enum):
     Feature.SLURM_DATABASE: [],
     Feature.CLUSTER_HEALTH_METRICS: ["us-iso"],
     Feature.CAPACITY_BLOCK: ["us-iso"],
+    Feature.NLB_SECURITY_GRROUP: ["us-iso"],
```
We should cover this in the unit test `test_Feature_validators.py::test_feature_region_validator` and `test_utils.py::test_is_feature_supported`

hanwen-pcluste replied:
Done

hanwen-cluster committed:
[integ-tests] Skip test_shared_home to in regions that do not support FSxOntap and FSxOpenZFS

Prior to this commit, the logic was wrong and didn't skip Ontap and OpenZFS if Lustre is supported

Signed-off-by: Hanwen <hanwenli@amazon.com>

hanwen-pcluste committed:
Merge branch 'develop' into developoct2

hanwen-pcluste committed:
Merge branch 'develop' into developoct2
